### PR TITLE
fix: high RPC usage in dispatcher

### LINKED ISSUE #278: Node make thousands of RPC requests when syncing
## 🙂 Expected behavior

The node should make just a few requests to the RPC provider when syncing to an application.
See the test cases for more details.

## 🫠 Actual behavior

The node makes thousands of requests to the RPC provider.

## 🧪 Minimal test case

This can be reproduced when starting the node for a DApp that was deployed to the  Sepolia testnet.
To start the node with this configuration, run `make docker-build` and `make docker-run-sepolia`.

When running the node with this configuration, it makes several requests to the RPC address and takes a long time to start processing the inputs.

## 🌎 Environment

Main branch. This doesn't seem like a regression.

### PR DESCRIPTION
**Pull request by gligneul**
Close #278 

I tested on Sepolia with Alchemy and it solves the issue.
Screenshot from 2024-01-24 18-15-19

### CODE REVIEW

**Review by endersonmaia — DISMISSED** [review]

File: offchain/types/src/foldables.rs [hunk 1]
```diff
@@ -151,14 +164,20 @@ async fn fetch_all_new_inputs<
...
     let contract = InputBox::new(*contract_address, Arc::clone(&provider));
 
+    // The application address is the first indexed parameter of the InputAdded event
+    let dapp_address = H160(dapp_address.to_owned().into());
+    let dapp_address_index = ValueOrArray::Value(Some(dapp_address.into()));
+
     // Retrieve `InputAdded` events
     let input_events = contract
         .input_added_filter()
+        .topic1(dapp_address_index)
```
Comment (GCdePaula) [inline offchain/types/src/foldables.rs:180]: I think you can just pass the `dapp_address`, no need for `ValueOrArray`:

```
    // Retrieve `InputAdded` events
    let input_events = contract
        .input_added_filter()
        .topic1(*dapp_address)
        .query_with_meta()
        .await
        .context("Error querying for input added events")?;
```

File: offchain/types/src/foldables.rs [hunk 2]
```diff
@@ -82,6 +87,7 @@ impl Foldable for InputBox {
         env: &StateFoldEnvironment<M, Self::UserData>,
         access: Arc<FoldMiddleware<M>>,
     ) -> Result<Self, Self::Error> {
+        let dapp_address = Arc::clone(&previous_state.dapp_address);
```
Comment (GCdePaula) [inline offchain/types/src/foldables.rs:90]: I think you could also check the bloom filter for `dapp_address`.

**Review by GCdePaula — APPROVED** [review]
LGTM! Left a few unnecessary comments.

File: offchain/types/src/foldables.rs (hunk 2 above)
Comment (gligneul) [reply offchain/types/src/foldables.rs:90]: @GCdePaula I added `fold_utils::contains_address(&block.logs_bloom, &dapp_address)` to the filter, but the dispatcher stopped reading inputs. Do you have a more specific suggestion?

Comment (gligneul) [reply offchain/types/src/foldables.rs:90]: Never mind; it worked with `fold_utils::contains_topic(&block.logs_bloom, &*dapp_address)`

**Review by GCdePaula — APPROVED** [review]
🚀🚀🚀

**Pull request merged**